### DISCUSSION TITLE
How to add the code item from Format menu to toolbar?

### DISCUSSION THREAD

**Discussion by Sarthak-Ghule** [tinymce/tinymce#7966] [General]
I want to add the code <kbd>code</kbd> item (not the sourcecode) from the Format menu to toolbar like the <kbd>bold</kbd>, <kbd>italic</kbd>, <kbd>subscript</kbd>, etc. 

![Screenshot (36)](https://user-images.githubusercontent.com/87312915/178154507-987b1c2f-dd0d-471f-9987-8c970da3b445.png)

From the [docs](https://www.tiny.cloud/docs/tinymce/6/basic-setup/#default-menu-controls), I found that we have to add `codeformat` in the toolbar string in the [toolbar configuration](https://www.tiny.cloud/docs/tinymce/6/basic-setup/#toolbar-configuration). But I don't see any result by doing so. How can I do that?

**Comment by TheSpyder** [comment]
Unfortunately this item is only registered as a menu, not a toolbar button.
https://github.com/tinymce/tinymce/blob/c12fb6af3b6b72886946dd3fb1bb1b8be249949f/modules/tinymce/src/themes/silver/main/ts/ui/core/SimpleControls.ts#L104-L108

I'll log a task to add a toolbar button for it but it would be easy to contribute if you'd like to make a PR (in the code I linked register a button with the same text, icon and action).

**Reply by ArthurRM** [reply]
Created a PR: 5208aa46ce292500bb745dbc82044be2354ead82

(have not made regular contributions to open source projects before so please enlighten me if I'm doing something wrong; the change was built & tested locally)

**Comment by he110-c0de** [comment]
Still exists in 2024
